homeassistant/components/uptimerobot/binary_sensor.py

```diff
@@ -21,69 +32,91 @@
 
-    for monitor in monitors["monitors"]:
-        devices.append(
+    async_add_entities(
+        [
             UptimeRobotBinarySensor(
-                api_key,
-                up_robot,
-                monitor["id"],
-                monitor["friendly_name"],
-                monitor["url"],
+                coordinator,
+                UptimeRobotBinarySensorEntityDescription(
+                    key=monitor["id"],
+                    name=monitor["friendly_name"],
+                    target=monitor["url"],
+                    device_class=DEVICE_CLASS_CONNECTIVITY,
+                ),
             )
-        )
-
-    add_entities(devices, True)
+            for monitor in coordinator.data["monitors"]
+        ],
+        True,
+    )
 
 
-class UptimeRobotBinarySensor(BinarySensorEntity):
+class UptimeRobotBinarySensor(BinarySensorEntity, CoordinatorEntity):
     """Representation of a Uptime Robot binary sensor."""
 
-    def __init__(self, api_key, up_robot, monitor_id, name, target):
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        description: UptimeRobotBinarySensorEntityDescription,
+    ) -> None:
         """Initialize Uptime Robot the binary sensor."""
-        self._api_key = api_key
-        self._monitor_id = str(monitor_id)
-        self._name = name
-        self._target = target
-        self._up_robot = up_robot
-        self._state = None
-
-    @property
-    def name(self):
-        """Return the name of the binary sensor."""
-        return self._name
-
-    @property
-    def is_on(self):
-        """Return the state of the binary sensor."""
-        return self._state
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return DEVICE_CLASS_CONNECTIVITY
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes of the binary sensor."""
-        return {ATTR_ATTRIBUTION: ATTRIBUTION, ATTR_TARGET: self._target}
-
-    def update(self):
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self.entity_description = description
+        self._attr_extra_state_attributes = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_TARGET: self.entity_description.target,
+        }
+
+    async def async_update(self):
         """Get the latest state of the binary sensor."""
-        monitor = self._up_robot.getMonitors(self._api_key, self._monitor_id)
-        if not monitor or monitor.get("stat") != "ok":
-            _LOGGER.warning("Failed to get new state")
-            return
-        status = monitor["monitors"][0]["status"]
-        self._state = 1 if status == 2 else 0
+        if monitor := get_monitor_by_id(
+            self.coordinator.data.get("monitors", []), self.entity_description.key
+        ):
+            self._attr_is_on = monitor["status"] == 2
+
+
+def get_monitor_by_id(monitors, monitor_id):
```

Comment:
This function can be replaced with a generator expression and `next` with default None.
```py
monitor = next(
    (monitor for monitor in monitors if monitor["id"] == monitor_id),
    None
)
```